### PR TITLE
GG-30905 Updated dependency versions

### PR DESCRIPTION
**Pull request by vsisko**

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -128,8 +128,8 @@
         <spark.hadoop.version>2.7.7</spark.hadoop.version>
         <spark.version>2.3.3</spark.version>
         <spring.boot.version>1.5.20.RELEASE</spring.boot.version>
-        <spring.data.version>1.13.14.RELEASE</spring.data.version> <!-- don't forget to update spring version -->
-        <spring.version>4.3.25.RELEASE</spring.version><!-- don't forget to update spring-data version -->
+        <spring.data.version>1.13.23.RELEASE</spring.data.version> <!-- don't forget to update spring version -->
+        <spring.version>4.3.29.RELEASE</spring.version><!-- don't forget to update spring-data version -->
         <spring.data-2.0.version>2.0.9.RELEASE</spring.data-2.0.version> <!-- don't forget to update spring-5.0 version -->
         <spring-5.0.version>5.0.17.RELEASE</spring-5.0.version><!-- don't forget to update spring-data-2.0 version -->
```
Comment (nva) [inline parent/pom.xml:134]: Should we upgrade spring-5.0.version? According https://snyk.io/vuln/SNYK-JAVA-ORGSPRINGFRAMEWORK-1009832 5.0.17.RELEASE is affected too

**Review by nva — CHANGES_REQUESTED** [review]

File: parent/pom.xml [hunk 2]
```diff
@@ -128,10 +128,10 @@
...
         <spark.version>2.3.3</spark.version>
         <spring.boot.version>1.5.20.RELEASE</spring.boot.version>
-        <spring.data.version>1.13.14.RELEASE</spring.data.version> <!-- don't forget to update spring version -->
-        <spring.version>4.3.25.RELEASE</spring.version><!-- don't forget to update spring-data version -->
-        <spring.data-2.0.version>2.0.9.RELEASE</spring.data-2.0.version> <!-- don't forget to update spring-5.0 version -->
-        <spring-5.0.version>5.0.17.RELEASE</spring-5.0.version><!-- don't forget to update spring-data-2.0 version -->
+        <spring.data.version>1.13.23.RELEASE</spring.data.version> <!-- don't forget to update spring version -->
+        <spring.version>4.3.29.RELEASE</spring.version><!-- don't forget to update spring-data version -->
+        <spring.data-2.0.version>2.0.14.RELEASE</spring.data-2.0.version> <!-- don't forget to update spring-5.0 version -->
+        <spring-5.0.version>5.0.18.RELEASE</spring-5.0.version><!-- don't forget to update spring-data-2.0 version -->
```
Comment (dmekhanikov) [inline parent/pom.xml:134]: 5.0.19 is already available. Can we use it instead?

**Review by dmekhanikov — CHANGES_REQUESTED** [review]

File: parent/pom.xml (hunk 1 above)
Comment (vsisko) [reply parent/pom.xml]: Updated spring 5.

File: parent/pom.xml (hunk 2 above)
Comment (vsisko) [reply parent/pom.xml]: Used 5.0.19 version.

**Comment by vsisko** [comment]
Reverted version of spring data 2.0. It produce an error for remove operation when list of removed entries are expected.
Created separated issue https://ggsystems.atlassian.net/browse/GG-30928

**Review by nva — APPROVED** [review]

**Review by wirtsleg — APPROVED** [review]

**Pull request merged**